src/analytics.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import fs from 'fs-extra'
+import { default as OS } from 'os'
+import type Config from 'cli-engine-config'
+import HTTP from 'cli-engine-command/lib/http'
+import Netrc from 'netrc-parser'
+import Output from 'cli-engine-command/lib/output'
+import URI from 'url'
+
+export default class AnalyticsCommand {
+  command: string
+  plugin: string
+  pluginVersion: string
+  version: string
+  platform: string
+  arch: string
+  start: any
+  config: Config
+
+  constructor (command: string, plugin: string, pluginVersion: string, config: Config) {
+    this.config = config
+    this.command = command
+    this.plugin = plugin
+    this.pluginVersion = pluginVersion
+    this.platform = OS.platform()
+    this.arch = OS.arch()
+  }
+
+  get version (): string {
+    return this.config.version
+  }
+
+  async recordEnd () {
+    if (AnalyticsCommand.skipAnalytics() || process.argv.length < 2) {
+      return
+    }
+    let analyticsJSON
+    if (fs.existsSync(AnalyticsCommand.analyticsPath(this.config))) {
+      let analyticsData = fs.readFileSync(AnalyticsCommand.analyticsPath(this.config), 'utf8')
+      analyticsJSON = JSON.parse(analyticsData)
+    } else {
+      analyticsJSON = {
+        'schema': 1,
+        'commands': [],
+        'user': 'not specified'
+      }
+      const netrcUser = await AnalyticsCommand.netrcLogin()
+      if (netrcUser) analyticsJSON['user'] = netrcUser
+    }
+    analyticsJSON['commands'].push({
+      command: this.command,
+      version: this.config.version,
+      platform: this.platform
+    })
+    fs.writeFileSync(AnalyticsCommand.analyticsPath(this.config), JSON.stringify(analyticsJSON), 'utf8')
+  }
+
+  static async submitAnalytics (config: Config) {
+    if (AnalyticsCommand.skipAnalytics()) return
+    const analyticsData = fs.readFileSync(this.analyticsPath(config), 'utf8')
+    let options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.parse(analyticsData)
+    }
+    let host = process.env['HEROKU_ANALYTICS_HOST']
+    if (host) {
+      const parsedHost: Object = URI.parse(host)
+      host = parsedHost.protocol + `//` + parsedHost.hostname + parsedHost.path
+      if (parsedHost.port) {
+      // flow$ignore
+        options['port'] = parsedHost.port
+      }
+    } else {
+      host = 'https://cli-analytics.heroku.com/record'
+    }
+    let out = new Output(config)
+    // flow$ignore
+    let http = new HTTP(out)
+    await http.post(host, options)
+  }
+
+  static skipAnalytics (config : Config) {
+    if (config.skipAnalytics) {
+      return true
+    } else if (AnalyticsCommand.netrcLogin() === false) {
+      return true
+    } else {
+      return false
+    }
+  }
+
+  static analyticsPath (config: Config): string {
+    return config.dirs.cache + '/analytics.json'
+  }
+
+  static async netrcLogin () {
+    // flow$ignore
```

Comment:
is this because of the return false? it should probably return undefined instead then this should work I think